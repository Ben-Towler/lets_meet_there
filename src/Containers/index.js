import Home from './Home/Home.container';
import FlightList from './FlightList/FlightList.container';

export {
  Home,
  FlightList
}