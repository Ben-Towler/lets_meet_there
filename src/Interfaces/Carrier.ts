export interface Carrier {
  CarrierId: number;
  Name: String;
}