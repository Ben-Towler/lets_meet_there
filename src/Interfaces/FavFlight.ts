interface FavLeg {
  CarrierIds: number[]; 
  OriginId: number;
  DestinationId: number; 
  DepartureDate: string;
}


export interface FavFlight {
  Direct: Boolean;
  InboundLeg: FavLeg;
  MinPrice: number;
  OutboundLeg: FavLeg;
  QuoteDateTime: string;
  QuoteId: number;
}