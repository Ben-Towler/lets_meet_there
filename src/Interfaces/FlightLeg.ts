export interface FlightLeg {
  CarrierIds: number[],
  OriginId: number,
  DestinationId: number,
  DepartureDate: string
}