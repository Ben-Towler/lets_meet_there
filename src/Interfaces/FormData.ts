export interface FormData {
  location?: string;
  destination?: string;
  startDate?: string;
  endDate?: string;
  name?: string;
}