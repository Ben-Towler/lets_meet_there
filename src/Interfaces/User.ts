export interface Favourites {
  userRequest: {
    origin: string,
    destination: string,
    outboundDate: string,
    inboundDate: string
  },
  friendRequest: {
    origin: string,
    destination: string,
    outboundDate: string,
    inboundDate: string
  }
}

export interface User {
  displayName: string,
  id: string,
  email: string,
  createdAt: {nanoseconds: number, seconds: number},
  favourites: Favourites[];
}
