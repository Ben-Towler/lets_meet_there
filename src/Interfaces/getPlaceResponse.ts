export interface GetPlaceResponse {
  PlaceId: string,
  PlaceName: string,
}