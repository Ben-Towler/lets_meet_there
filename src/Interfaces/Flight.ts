import {FlightLeg} from './FlightLeg';

export interface Flight {
  QuoteId: number;
  MinPrice: number;
  Direct: Boolean,
  OutboundLeg: FlightLeg,
  InboundLeg: FlightLeg,
  QuoteDateTime: string,
}



