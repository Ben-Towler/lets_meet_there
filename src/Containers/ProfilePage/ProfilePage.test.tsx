import React from 'react'
import {render} from '@testing-library/react'
import {ProfilePage} from 'Containers';
import { userMocks } from 'Mocks/data.mock';
import { Favourites } from 'Interfaces/User';
 
const mocks = {
  noDataUser: {
    displayName: 'Andrew',
    email: '',
    createdAt: {nanoseconds: 0, seconds: 0},
    id: '',
    favourites: [] as Favourites[]
  }
};

describe('test the profile page load', () => {
  it('should match snapshot and render correctly', () => {
    const { asFragment } = render(<ProfilePage user={userMocks} />);
    
    expect(asFragment()).toMatchSnapshot();
  })
  
  it('Should display the displayName', () => {
    const { queryByText } = render(<ProfilePage user={userMocks} />);
    const display = queryByText('Andrew');
    expect(display).toBeInTheDocument();
  })
  
  it('should tell the user when there are no favourite quotes', () => {
    const { queryByText } = render(<ProfilePage user={mocks.noDataUser} />);
    const noQuotes = queryByText('You have no quotes favourited');
    expect(noQuotes).toBeInTheDocument();
  })

  it('should render differently when favourite data exists', () => {
    const { queryByText } = render(<ProfilePage user={userMocks} />);
    const noQuotes = queryByText('You have no quotes favourited');
    expect(noQuotes).not.toBeInTheDocument();
  })

})

