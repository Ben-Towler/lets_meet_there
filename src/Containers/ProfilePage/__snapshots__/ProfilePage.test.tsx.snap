// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the profile page load should match snapshot and render correctly 1`] = `
<DocumentFragment>
  <div
    class="spinner-overlay"
  >
    <div
      class="spinner-container"
    />
  </div>
</DocumentFragment>
`;
